### ISSUE TITLE
bug in dividing polynomials by module

### ISSUE THREAD

**Issue by MariaMsu** [sympy/sympy#20397]
```
In [10]: from sympy import prem, pquo
    ...: from sympy.abc import x
    ...: !pip show sympy
Name: sympy
Version: 1.6.2
Summary: Computer algebra system (CAS) in Python
Home-page: https://sympy.org
Author: SymPy development team
Author-email: sympy@googlegroups.com
License: BSD
Location: /home/maria/Desktop/env_p3.8/lib/python3.8/site-packages
Requires: mpmath
Required-by: 

In [11]: def count(f, g, modulus):
    ...:     print(f"quotient = {pquo(f, g, modulus=modulus)}")
    ...:     print(f"remainder= {prem(f, g, modulus=modulus)}")
    ...: 
    ...: count(x**2+x+3, 2*x+2, modulus=7)
quotient = 2*x
remainder= -2
```
Вut right answer is:
```
quotient = 4*x 
remainder= 3  # or =-4, assuming that (-4 mod 7) = 3
```
 That is, the output is not multiplied by ``2`` - the coefficient of the second polynomial

**Comment by shivangdubey** [comment]
What shall be done to solve this?

**Comment by uver29** [comment]
Can anyone tell, what is Polynomial pseudo-quotient and remainder? I read the documentation but I didn't understand it.

**Comment by MariaMsu** [comment]
handwritten solution:
![image](https://user-images.githubusercontent.com/35924838/98469365-2c0f4d00-21f0-11eb-88d5-7a2925f6c556.png)

**Comment by MariaMsu** [comment]
> What shall be done to solve this?

I did not understand the library's backend. Probably the coefficient of the divisor's prime degree is lost

**Comment by wsdookadr** [comment]
So, I stepped through the code with ipdb and realized that [dup_prem](https://github.com/sympy/sympy/blob/78891c0/sympy/polys/densearith.py#L1053) is being called and [dup_pquo](https://github.com/sympy/sympy/blob/78891c0/sympy/polys/densearith.py#L1098) .

**Comment by int-e** [comment]
I don't believe this is a bug. `prem` and `pquo` do *pseudo-division* of polynomialis which differs from normal division in that it can be computed without division in the base ring, while producing a remainder of degree less than the divisor. This is achieved, conceptually, by multiplying the first operand by a sufficiently large power of the leading coefficient of the second argument with exponent. It is useful for gcd computations, see [Wikipedia](https://en.wikipedia.org/wiki/Pseudo-remainder#Pseudo-remainder_sequences).

For working with polynomials over a field, `rem` and `quo` (or `div`, which combines both) do the right thing:
```
>>> from sympy import rem, quo
>>> from sympy.abc import x
>>> f = x**2 + x + 3
>>> g = 2*x+2
>>> quo(f,g,modulus=7)
-3*x
>>> rem(f,g,modulus=7)
3
```

The documentation of `prem` and `pquo` could be improved though... at the very least a warning that pseudo-division is not the same as ordinary division and that for most uses, `rem`, `quo` and `div` are what people need. Ideally, it should spell out what pseudo-division does. Here's a start:

```
Pseudo-division of ``f`` and ``g`` is basically the same as
dividing ``f * LC(g) ** (degree(f) - degree(g) + 1)`` by ``g``,
but is implemented without performing any divisions by ``LC(g)``.
```

**Comment by sylee957** [comment]
When anyone attempt to change the documentation. make sure that the documentation change is symmetric for all `pdiv`, `pquo`, `prem` for all the functions/methods.
